sdk/core/abort-controller/src/AbortController.ts

```diff
@@ -69,7 +69,13 @@ export class AbortController {
    */
   constructor(...parentSignals: AbortSignalLike[]);
   constructor(parentSignals?: any) {
-    this._signal = new AbortSignal();
+    if (typeof window !== "undefined" && typeof window.AbortController === "function") {
```

Comment:
Should we check `self` instead of `window` in case we're running inside a webworker?

Reply:
I think you're right... we need to abstract out these platform checks in a better way. We're checking `window` in a lot of places outside of this library.